connectors/src/resources/slack_configuration_resource.ts

```diff
@@ -0,0 +1,216 @@
+        connectorId: connectorId,
+      },
+      { transaction }
+    );
+  }
+
+  static async fetchByConnectorId(connectorId: ModelId) {
+    const blob = await this.model.findOne({
+      where: {
+        connectorId: connectorId,
+      },
+    });
+    if (!blob) {
+      return null;
+    }
+
+    // Use `.get` to extract model attributes, omitting Sequelize instance metadata.
+    return new this(this.model, blob.get());
+  }
+
+  static async fetchByActiveBot(slackTeamId: string) {
+    const blob = await this.model.findOne({
+      where: {
+        slackTeamId,
+        botEnabled: true,
+      },
+    });
+    if (!blob) {
+      return null;
+    }
+
+    // Use `.get` to extract model attributes, omitting Sequelize instance metadata.
+    return new this(this.model, blob.get());
+  }
+
+  async isBotWhitelisted(botName: string | string[]): Promise<boolean> {
+    return !!(await SlackBotWhitelistModel.findOne({
+      where: {
+        connectorId: this.id,
+        botName: botName,
+      },
+    }));
+  }
+
+  static async listAll() {
+    const blobs = await SlackConfigurationResource.model.findAll({});
+
+    return blobs.map(
+      // Use `.get` to extract model attributes, omitting Sequelize instance metadata.
+      (b) => new SlackConfigurationResource(this.model, b.get())
+    );
+  }
+
+  static async listForTeamId(
+    slackTeamId: string
+  ): Promise<SlackConfigurationResource[]> {
+    const blobs = await this.model.findAll({
+      where: {
+        slackTeamId,
+        botEnabled: true,
+      },
+    });
+
+    return blobs.map(
+      (b) => new SlackConfigurationResource(this.model, b.get())
+    );
+  }
+
+  async enableBot(): Promise<Result<undefined, Error>> {
+    const otherSlackConfigurationWithBotEnabled =
+      await SlackConfigurationModel.findOne({
+        where: {
+          slackTeamId: this.slackTeamId,
+          botEnabled: true,
+        },
+      });
+    if (
+      otherSlackConfigurationWithBotEnabled &&
+      otherSlackConfigurationWithBotEnabled.id !== this.id
+    ) {
+      logger.error(
+        {
+          slackTeamId: this.slackTeamId,
+        },
+        "Another Dust workspace has already enabled the slack bot for your Slack workspace."
+      );
+      return new Err(
+        new Error(
+          "Another Dust workspace has already enabled the slack bot for your Slack workspace."
+        )
+      );
+    }
+    await this.model.update(
+      { botEnabled: true },
+      {
+        where: {
+          id: this.id,
+        },
+      }
+    );
+
+    return new Ok(undefined);
+  }
+
+  async disableBot(): Promise<Result<undefined, Error>> {
+    await this.model.update(
+      { botEnabled: false },
+      {
+        where: {
+          id: this.id,
+        },
+      }
+    );
+
+    return new Ok(undefined);
+  }
+
+  async setWhitelistedDomains(domain: string[]) {
+    await this.model.update(
+      { whitelistedDomains: domain },
```

Comment:
I'd be in favor of moving that to its own table so that everything is consistent